### DISCUSSION TITLE
native logger integration / dep loglevel config?

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [aws-powertools/powertools-lambda-python#819] [Q&A] [answered]
Python comes with some logging niceties [built-in](https://docs.python.org/3/howto/logging.html), among which the ability to use yaml-based configuration for logging. this facilitates e.g. the configuration of logging levels of dependencies.
this sounds nice, but i'm not sure how to achieve something similar with powertools's logger. is this something that is currently possible, or might this be desirable as a potential enhancement?

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! We'll come back to you as soon as we can.

**Comment by heitorlessa** [comment]
hey @KiaraGrouwstra thank you for creating it - apologies for the delay, just got back from a long vacation.

For setting the log level, you can use either `Logger(level="DEBUG")` or simply the environment variable `LOG_LEVEL` as per our docs: https://awslabs.github.io/aws-lambda-powertools-python/latest/core/logger/#getting-started

Child loggers (`Logger(child=true)`) have bi-directional sync, which means if you change the logger it'll replicate both ways, for example if you add a new key: `logger.append_keys(payment_id="123")`

We don't plan to add a YAML based config as you don't need to configure it like you do the traditional logger, you simply do `logger = Logger()`, and it just works. You can opt-in for other features like setting correlation ID, log sampling, but you'd want to do this conditionally, not statically.

Does that help?

I'm more than happy to clarify any questions you might have.

**Reply by KiaraGrouwstra** [reply]
that sounds good, I'll try -- thank you!

**Comment by KiaraGrouwstra** [comment]
hi,
this issue i'm concerned about is about dependencies.
i know that for my own code i can control which logging level of messages will be printed.
the issue is, i'm consuming other pip packages that log things as well, and in some cases turn out buggy.
in such cases, i would like to be able to print debug-level info for such a broken package.
that is something that by default python logging can achieve. however, as i currently understand it, since the powertools logging library does not hook into this built-in way of logging, it lacks that functionality.
is this understanding correct?

**Comment by heitorlessa** [comment]
Ah! I totally misunderstood, sorry.

You can get the namespace logger and set the level as you normally would
with any logging that used std logging[1].

Could you share the workflow you have in mind so I can share a snippet with
you?

I don’t wanna assume you want to propagate down, upwards, or in another way
I might not be thinking.

-----------

> Edited to add an initial snippet in case it's helpful.

[1] Creating a Logger named `booking` and setting their log level to DEBUG elsewhere in the code

```python
# app.py
from aws_lambda_powertools import Logger

logger = Logger(service="booking")

# somewhere in the code
import logging

booking_logger = logging.getLogger("booking")  # we name the logger using service name value
booking_logger.setLevel("DEBUG")
```

**Comment by heitorlessa** [comment]
Checking in to see if this helped, and whether we could have more details on the issue report about the workflow to address this properly

**Comment by KiaraGrouwstra** [comment]
hi there,
i'm not sure i fully understood the proposed solution.
i wanna use this library for logging, while a buggy dependency of mine is not aware of the existence of this library. moreover, given this is an external library i'm consuming, i do not have direct control over the code it uses.
if i use native logging for my own application, i can configure the logging level for my buggy dependency using config files.
what would be the equivalent when for my own application i wish to log using `aws-lambda-powertools`?

**Comment by heitorlessa** [comment]
Hey Kiara - I’m not sure whether it helps on your use case but I’m gonna
add a new method in Logger to copy its configuration like formatting, JSON,
log level to all loggers in an application.

This could help in situations where you want other libraries - e.g. boto3 -
to use the same Log Level & output consistently.

When you have a chance, I’d love to see what the workflow you have in mind
for your problem since I haven’t used file-based configuration much before.


About the new feature

https://github.com/awslabs/aws-lambda-powertools-python/discussions/799#discussioncomment-1587777

**Comment by heitorlessa** [comment, marked as answer]
Hey Kiara - We finally have this as a standalone function now (1.24.1) - please do let us know if this helps out. You can also explicitly configure which logger you want to copy Powertools Logger config to, and also set a different Log Level if you want to -- Omitting these will copy Logger structured config to all imported loggers.

Release 1.24.1: https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.24.1

**Reply by KiaraGrouwstra** [reply]
that looks great, can't wait to try this out :relaxed:
